tests/unit/NetworkTest.js

```diff
@@ -209,3 +221,45 @@ test('consecutive API calls eventually succeed when authToken is expired', () =>
             expect(chatList).toEqual(TEST_CHAT_LIST);
         });
 });
+
+test('retry network request if auth and credentials are not read from Onyx yet', () => {
+    Onyx.addDelayToConnectCallback(3000);
+
+    // Reset isReady for this unit test, because Onyx.clear() in beforeEach resolves the
+    // Onyx.connect callbacks with null values
+    Network.setIsReady(false);
+
+    // Set values to trigger Onyx.connect callbacks
+    Onyx.merge(ONYXKEYS.CREDENTIALS, {});
+    Onyx.merge(ONYXKEYS.SESSION, {});
```

Comment:
I'm confused about why we are merging empty objects here and what the initial state of these values should be.

Reply:
e.g. I think the bug we are observing has the values for `authToken` and `credentials` already existing, but just taking a long time to be read from storage. So I'm unsure what we are trying to simulate by merging in the empty object..

Reply:
I'm merging an empty object just to force the `Onyx.connect` callbacks registered in `API.js` (which kind of workaround the problem of the previous [comment](https://github.com/Expensify/App/pull/6183#discussion_r743222649)). And it's an empty object because in this unit test we only want to verify that a network request is retried when the `authToken` (which can be non-existent, valid, or expired) and `credentials` haven't been read from the storage.

Also, I think the initial values are empty because of the `Onyx.clear` call in `beforeEach`, that's why I thought made sense to merge only an empty object in both.

Reply:
I think I'm following... The comment does explain that we want trigger the callback, but I was unsure why we chose to merge an empty object into these keys to do that. Thinking about it some more I guess the answer is... it's just _something_ that won't affect the value and will trigger an update. And we are having difficulty creating a situation where `Onyx.connect()` has not yet run.

Reply:
I added a comment to explain this better and why we merge an empty object 👍🏽

Reply:
Great thanks!